### PR TITLE
holdings: add new holdings types

### PR DESCRIPTION
**Pull request by BadrAly**
Three types of holdings are now available. The serial holdings type is used by the journal documents
i.e. documents with serial as RDA issuance mode.
The electronic holdings type is used by the ebooks documents, i.e. harvested documents.
The standard holdings type is used by all other type of documents.

* Disallows the creation/update/deletion of non-standard type of holdings.
* Adds the number of holdings as a condition to block a document deletion.
* Fixes the holdings json schema required fields and description.
* Regenerates the big and small data files because of the changes made to the holdings json schema.
* Defines better the patterns examples, to make clear difference between the enumeration and chronology captions.

Co-Authored-by: Aly Badr <aly.badr@rero.ch>

## Why are you opening this PR?

https://tree.taiga.io/project/rero21-reroils/task/1396?kanban-status=1224894

## How to test?

`bootstrap, setup`
UI tests will be available in : https://tree.taiga.io/project/rero21-reroils/task/1375?kanban-status=1224894

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?
- [ ] Extracted translations?

### CODE REVIEW

File: rero_ils/modules/holdings/api.py [hunk 1]
```diff
@@ -98,22 +98,30 @@ def extended_validation(self, **kwargs):
...
-        is_issuance = document.get('issuance') == 'rdami:1003'
-        if (is_issuance and not is_serial) or (not is_issuance and is_serial):
-            return False
-        return True
+        document = Document.get_record_by_pid(self.document_pid)
+        is_serial = self.holdings_type == 'serial'
+        is_electronic = self.holdings_type == 'electronic'
+        is_issuance = document.dumps().get('issuance') == 'rdami:1003'
+        return not((is_issuance ^ is_serial)
+                   or(document.harvested ^ is_electronic))
```
Comment (zannkukai) [inline rero_ils/modules/holdings/api.py:110]: Yeeeah I discovered a new operator !

Comment (BadrAly) [reply rero_ils/modules/holdings/api.py]: Thanks goes to Peter

File: rero_ils/modules/documents/api.py [hunk 2]
```diff
@@ -109,6 +109,12 @@ def get_number_of_items(self):
         return ItemsSearch().filter(
             'term', document__pid=self.pid).source().count()
 
+    def get_number_of_holdings(self):
+        """Get number of holdings for document."""
+        from ..holdings.api import HoldingsSearch
+        return HoldingsSearch().filter(
+            'term', document__pid=self.pid).source().count()
```
Comment (jma) [inline rero_ils/modules/documents/api.py:116]: source here seems useless

File: rero_ils/modules/holdings/api.py [hunk 3]
```diff
@@ -98,22 +98,31 @@ def extended_validation(self, **kwargs):
 
         Ensures that holdings of type serials are created only on
         journal documents i.e. serial mode_of_issuance documents.
+
+        Ensures that holdings of type electronic are created only on
+        ebooks documents i.e. harvested documents.
```
Comment (jma) [inline rero_ils/modules/holdings/api.py:103]: please add @returns and @params

File: rero_ils/modules/holdings/api.py [hunk 4]
```diff
@@ -98,22 +98,31 @@ def extended_validation(self, **kwargs):
...
-        is_serial = self.get('holdings_type') == 'serial'
-        is_issuance = document.get('issuance') == 'rdami:1003'
-        if (is_issuance and not is_serial) or (not is_issuance and is_serial):
-            return False
-        return True
+        document = Document.get_record_by_pid(self.document_pid)
+        is_serial = self.holdings_type == 'serial'
+        is_electronic = self.holdings_type == 'electronic'
+        is_issuance = document.dumps().get('issuance') == 'rdami:1003'
+        return not(
```
Comment (jma) [inline rero_ils/modules/holdings/api.py:109]: probably better to write:
```python
return (is_issuance and is_serial) ^ (document.harvested and is_electronic)
```

File: rero_ils/modules/holdings/api.py [hunk 5]
```diff
@@ -232,8 +241,12 @@ def get_number_of_items(self):
         return results
 
     def get_links_to_me(self):
-        """Get number of links."""
+        """Get links that can block the holding deletion.
+
+        Attached items to a holding record blocks the deletion.
```
Comment (jma) [inline rero_ils/modules/holdings/api.py:246]: please add @returns

File: rero_ils/modules/items/api.py [hunk 6]
```diff
@@ -267,15 +267,15 @@ def _item_build_org_ref(cls, data):
         data['organisation'] = org_ref
 
     def item_link_to_holding(self):
```
Comment (jma) [inline rero_ils/modules/items/api.py:269]: link_item_to_holding?

File: tests/ui/ebooks/test_ebooks_receivers.py [hunk 7]
```diff
@@ -37,6 +37,7 @@ def test_publish_harvested_records(app, ebooks_1_xml, ebooks_2_xml,
 
     assert Document.count() == 2
     doc1 = Document.get_record_by_pid('1')
+
```
Comment (jma) [inline tests/ui/ebooks/test_ebooks_receivers.py:40]: usefull?

File: rero_ils/modules/holdings/api.py (hunk 4 above)
Comment (BadrAly) [reply rero_ils/modules/holdings/api.py:115]: Did not work! can you take a deeper look because now I am worried that my logic is not correct.

Comment (jma) [reply rero_ils/modules/holdings/api.py:115]: You can leave at it is.

**Pull request merged**